EventFilter/CTPPSRawToDigi/src/RawDataUnpacker.cc

```diff
@@ -452,7 +458,16 @@ int RawDataUnpacker::processOptoRxFrameSampic(const word *buf,
     }
     // save frame to output
     frame.setPresenceFlags(1);
-    fc->Insert(fp, frame);
+    if ((optoRxId >= FEDNumbering::MINTotemT2FEDID && optoRxId <= FEDNumbering::MAXTotemT2FEDID)) {
+       if (verbosity>0) 
+          edm::LogWarning("Totem")<<"T2 Frame Positions created: "<<fp2a<<"/"<<fp2b<<std::endl;
+
+      fc->Insert(fp2a, frame); //Fill twice with the mapping for both payloads packed into the same frame, will match only once for each
```

Comment:
what is the convention here ? this seems a bit strange to include the frame twice

Reply:
Hi. It's because while for the non-T2 frame the triplet of values read in from the VFAT frame headers uniquely define a hwID from the mapping that we can look up, in the case of T2, CH0 and CH1 within the same payload frame will have completely different hwID, that both have to be looked up. I just tested that commenting out fp2b here meant no T2Digi were created from CH1, due to no hwID match being found.